### PR TITLE
z_bg_hidan_sima OK

### PR DESCRIPTION
**Pull request by mzxrules**

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c [hunk 1]
```diff
@@ -62,28 +68,243 @@ static ColliderJntSphInit D_8088F1E8 = {
...
+}
+
+void func_8088E90C(BgHidanSima* this) {
+    s32 pad[2];
+    f32 cos;
+    f32 sin;
+    s32 i;
+
+    cos = Math_CosS(this->dyna.actor.world.rot.y + 0x8000);
+    sin = Math_SinS(this->dyna.actor.world.rot.y + 0x8000);
```
Comment (petrie911) [inline src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c:199]:
```suggestion
    f32 cos = Math_CosS(this->dyna.actor.world.rot.y + 0x8000);
    f32 sin = Math_SinS(this->dyna.actor.world.rot.y + 0x8000);
    s32 i;
```

File: src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c [hunk 2]
```diff
@@ -62,28 +68,243 @@ static ColliderJntSphInit D_8088F1E8 = {
...
+    this->dyna.actor.world.pos.z = Math_CosS(this->dyna.actor.world.rot.y) * temp + this->dyna.actor.home.pos.z;
+    if (this->timer == 0) {
+        this->timer = 20;
+        this->actionFunc = func_8088E760;
+    }
+    func_8002F974(&this->dyna.actor, NA_SE_EV_FIRE_PILLAR - SFX_FLAG);
+}
+
+void func_8088E90C(BgHidanSima* this) {
+    s32 pad[2];
```
Comment (petrie911) [inline src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c:193]: surprised so many pads are needed. Could `i` replace one of them?

File: src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c [hunk 3]
```diff
@@ -62,28 +68,243 @@ static ColliderJntSphInit D_8088F1E8 = {
...
+        s32 temp;
+        if (this->dyna.actor.world.rot.y == this->dyna.actor.shape.rot.y) {
+            temp = this->timer;
+        } else {
+            temp = this->timer + 80;
+        }
+        if (this->actionFunc == func_8088E7A8) {
+            temp += 20;
+        }
+        this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y - ((1.0f - cosf((f32)temp * (M_PI / 20))) * 5.0f);
```
Comment (petrie911) [inline src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c:225]:
```suggestion
        this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y - ((1.0f - cosf(temp * (M_PI / 20))) * 5.0f);
```
This cast should be unnecessary.

File: src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c [hunk 4]
```diff
@@ -62,28 +68,243 @@ static ColliderJntSphInit D_8088F1E8 = {
...
+    phi_s5 = (60 - this->timer) >> 1;
+    if (phi_s5 >= 4) {
+        phi_s5 = 3;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Sima/func_8088E518.s")
+    v0 = 3 - (this->timer >> 1);
+    if (v0 < 0) {
+        v0 = 0;
+    }
```
Comment (petrie911) [inline src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c:254]: Can these be CLAMP?

File: src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c [hunk 5]
```diff
@@ -62,28 +68,243 @@ static ColliderJntSphInit D_8088F1E8 = {
...
+}
+
+Gfx* func_8088EB54(GlobalContext* globalCtx, BgHidanSima* this, Gfx* disp) {
+    MtxF mtxF;
+    s32 phi_s5;
+    s32 s3;
+    s32 v0;
+    f32 cos;
+    f32 sin;
+    s32 pad[2];
```
Comment (petrie911) [inline src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c:240]: It's possible these could be used as temps to tidy up the later calculations.

File: src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c [hunk 6]
```diff
@@ -62,28 +68,243 @@ static ColliderJntSphInit D_8088F1E8 = {
...
+    s32 pad;
+
+    this->actionFunc(this, globalCtx);
+    if (this->dyna.actor.params != 0) {
+        s32 temp;
+        if (this->dyna.actor.world.rot.y == this->dyna.actor.shape.rot.y) {
+            temp = this->timer;
+        } else {
+            temp = this->timer + 80;
+        }
```
Comment (petrie911) [inline src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c:221]:
```suggestion
        s32 temp = (this->dyna.actor.world.rot.y == this->dyna.actor.shape.rot.y) ? this->timer : (this->timer + 80)
```
assuming it matches and it's a way we want to write it.

**Review by petrie911 — COMMENTED** [review]
huh, `func_8088EB54` is quite the mess.

File: src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c [hunk 7]
```diff
@@ -62,28 +68,243 @@ static ColliderJntSphInit D_8088F1E8 = {
...
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Sima/BgHidanSima_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Sima/BgHidanSima_Destroy.s")
+Gfx* D_8088F1FC[] = { 0x06015D20, 0x06016120, 0x06016520, 0x06016920, 0x06016D20, 0x06017120, 0x06017520, 0x06017920 };
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c:79]: Those are textures. It should be `u64* D_8088F1FC[]`

**Review by AngheloAlf — CHANGES_REQUESTED** [review]
just a small change

File: src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c [hunk 8]
```diff
@@ -62,28 +68,243 @@ static ColliderJntSphInit D_8088F1E8 = {
...
+        }
+        this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y - ((1.0f - cosf((f32)temp * (M_PI / 20))) * 5.0f);
+        if (this->actionFunc == func_8088E7A8) {
+            func_8088E90C(this);
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+    }
+}
+
+Gfx* func_8088EB54(GlobalContext* globalCtx, BgHidanSima* this, Gfx* disp) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c:233]: Pretty minor but we usually call this kind of argument `gfx` (see `z_rcp.c` functions for instance)
```suggestion
Gfx* func_8088EB54(GlobalContext* globalCtx, BgHidanSima* this, Gfx* gfx) {
```

File: src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c (hunk 5 above)
Comment (mzxrules) [reply src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c:240]: true. I don't feel like doing a documentation pass now, so i'd rather leave them open for whoever does it.

File: src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c (hunk 4 above)
Comment (mzxrules) [reply src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c]: CLAMP works only if I assign to the var first

**Review by Roman971 — APPROVED** [review]

**Review by fig02 — APPROVED** [review]

**Pull request merged**